app/components/autocomplete_selector/autocomplete_selector.tsx

```diff
@@ -15,29 +14,42 @@ import {preventDoubleTap} from '@utils/tap';
 import {makeStyleSheetFromTheme, changeOpacity} from '@utils/theme';
 import {ViewTypes} from '@constants';
 import {goToScreen} from '@actions/navigation';
+import {ActionResult} from '@mm-redux/types/actions';
+import {Channel} from '@mm-redux/types/channels';
+import {DialogOption} from '@mm-redux/types/integrations';
+import {UserProfile} from '@mm-redux/types/users';
+import {Theme} from '@mm-redux/types/preferences';
 
-export default class AutocompleteSelector extends PureComponent {
-    static propTypes = {
-        actions: PropTypes.shape({
-            setAutocompleteSelector: PropTypes.func.isRequired,
-        }).isRequired,
-        getDynamicOptions: PropTypes.func,
-        label: PropTypes.string,
-        placeholder: PropTypes.string.isRequired,
-        dataSource: PropTypes.string,
-        options: PropTypes.arrayOf(PropTypes.object),
-        selected: PropTypes.object,
-        optional: PropTypes.bool,
-        showRequiredAsterisk: PropTypes.bool,
-        teammateNameDisplay: PropTypes.string,
-        theme: PropTypes.object.isRequired,
-        onSelected: PropTypes.func,
-        helpText: PropTypes.node,
-        errorText: PropTypes.node,
-        roundedBorders: PropTypes.bool,
-        disabled: PropTypes.bool,
-    };
+type Selection = DialogOption | Channel | UserProfile | DialogOption[] | Channel[] | UserProfile[];
+
+type Props = {
+    actions: {
+        setAutocompleteSelector: (dataSource: any, onSelect: any, options: any, getDynamicOptions: any) => Promise<ActionResult>,
+    },
+    getDynamicOptions?: (term: string) => Promise<ActionResult>,
+    label?: string,
+    placeholder?: string,
```

Comment:
This prop changed from required to optional. This seems like it should be typed as optional but just pointing it out

Reply:
I don't remember 100% if I had any other motivation, but yes, this value should not be required.